.github/workflows/test.yml

```diff
@@ -0,0 +1,30 @@
+name: test
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+permissions:
+  actions: read
+  contents: read
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v2
```

Comment:
You'll want to use `pnpm/action-setup@v3` as v2 is node v16 which is now deprecated on GHA. You can see the warning in the GHA results.